RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate subscriptions/room to TS

## Proposed changes
Migrate subscriptions/room to TS.

## Issue(s)	
N/A

## How to test or reproduce
Open a chat:
* check if the messages are loading;
* check if the messages are updated;
* check the threads;
* check if leave a room is still working;

Check if the list of chats is still working;

## Screenshots
N/A

## Types of changes




## Checklist


- [] I have added tests that prove my fix is effective or that my feature works (if applicable)

## Further comments
N/A
